### PR TITLE
[WIP] WatchKeys is sometimes latent

### PR DESCRIPTION
**Pull request by rudle**
the <-rcKeyWatch block here is potentially slow and upstream code blocks on
writing to this error channel

I am open to an all encompassing discussion about the sanity of error channels.

### CODE REVIEW

File: pkg/rc/farm.go [hunk 1]
```diff
@@ -133,6 +133,15 @@ func (rcf *Farm) mainLoop(quit <-chan struct{}) {
 
 	rcKeyWatch, rcErr := rcf.rcStore.WatchRCKeysWithLockInfo(subQuit, rcf.rcWatchPauseTime)
 
+	go func(errCh chan error) {
+		select {
```
Comment (mpuncel) [inline pkg/rc/farm.go]: i'm guessing you want a `for { ... }` around this

Comment (rudle) [inline pkg/rc/farm.go]: ofc

File: pkg/rc/farm.go [hunk 2]
```diff
@@ -133,6 +133,15 @@ func (rcf *Farm) mainLoop(quit <-chan struct{}) {
 
 	rcKeyWatch, rcErr := rcf.rcStore.WatchRCKeysWithLockInfo(subQuit, rcf.rcWatchPauseTime)
 
+	go func(errCh chan error) {
```
Comment (mpuncel) [inline pkg/rc/farm.go]: may as well put a direction on the channel here

File: pkg/rc/farm.go [hunk 3]
```diff
@@ -133,6 +133,17 @@ func (rcf *Farm) mainLoop(quit <-chan struct{}) {
 
 	rcKeyWatch, rcErr := rcf.rcStore.WatchRCKeysWithLockInfo(subQuit, rcf.rcWatchPauseTime)
 
+	go func(errCh <-chan error) {
+		for {
+			select {
+			case <-quit:
+				return
+			case err := <-rcErr:
```
Comment (mpuncel) [inline pkg/rc/farm.go]: don't think it matters but just noticed this is using the variable from outer scope rather than the arg

**Pull request merged**